cm-vault/vault-config/main.tf

```diff
@@ -42,3 +42,45 @@ resource "vault_pki_secret_backend_root_cert" "ca" {
   key_bits             = 2048
   exclude_cn_from_sans = true
 }
+
+# AppRole Authentication Method
+resource "vault_auth_backend" "approle" {
+  count = var.vault_authentication_mechanism == "approle" ? 1 : 0
+
+  type = "approle"
+}
+
+resource "vault_approle_auth_backend_role" "auth_role" {
+  count = var.vault_authentication_mechanism == "approle" ? 1 : 0
+
+  backend        = vault_auth_backend.approle[0].path
+  role_name      = "auth-role"
+  token_policies = ["default", "dev", "prod"]
+}
+
+resource "vault_approle_auth_backend_role_secret_id" "auth_role_secret_id" {
+  count = var.vault_authentication_mechanism == "approle" ? 1 : 0
+
+  backend   = vault_auth_backend.approle[0].path
+  role_name = vault_approle_auth_backend_role.auth_role[0].role_name
+}
+
+resource "vault_policy" "approle-pki-policy" {
+  count = var.vault_authentication_mechanism == "approle" ? 1 : 0
+
+  name   = "dev"
```

Comment:
Suggestion: name it something like 'cert-manager-issuer-policy` or perhaps just `default` as I think we will not have dev/prod policies here.